base/abstractarray.jl

```diff
@@ -392,7 +392,12 @@ julia> stride(A,3)
 12
 ```
 """
-stride(A::AbstractArray, k::Integer) = strides(A)[k]
+function stride(A::AbstractArray, k::Integer)
+    st = strides(A)
+    k < 1 && return 1
+    k ≤ ndims(A) && return st[k]
+    length(A)
```

Comment:
This doesn't seem to be the right answer, but I'm not sure what the right answer is.  Is it `sum(st .* size(A))`?

I know, it has a bit of that "does it matter what number is going to be multiplied by zero" rhetorical question quality, but this seems to be a better answer.

Reply:
Er, no, that's not quite it either.  Is it the absolute maximum of `st .* size(A)` (but preserving its sign)?  Even that leaves a bit to be desired as an abstract behavior... for example, here's how SubArray currently works:

```Julia
julia> v = @view rand(100)[1:3:end];

julia> stride(v, 1), stride(v, 2) # This is the answer the abstract behavior I proposed would give
(3, 102)

julia> m = @view rand(100, 1)[1:3:end, :];

julia> stride(m, 1), stride(m, 2) # This seems like a "better" answer:
(3, 100)
```

Reply:
I feel like an overestimate here is somehow more defensible than an underestimate as it pushes you _out of bounds_.  Here are a few more tricky n > ndims cases:

```Julia
julia> m  = @view rand(12, 24, 60)[1:3:end, end:-4:1, 2:-1:1];

julia> strides(m)
(3, -48, -288)

julia> stride(m, 4) # This seems reasonable; increasing this would jump out-of-bounds of the view
-576

julia> m2  = @view rand(12, 24, 60, 1)[1:3:end, end:-4:1, 2:-1:1, :];

julia> strides(m2) # this would _definitely_ jump out of bounds (that's the length of the parent, and we're starting in the middle!)
(3, -48, -288, 17280)

julia> p = PermutedDimsArray(m, (3,2,1));

julia> strides(p)
(-288, -48, 3)

julia> stride(p, 4)
ERROR: BoundsError: attempt to access (-288, -48, 3)
  at index [4]

julia> vpv = @view p[1:2, 1:2, 1:2]

julia> strides(vpv)
(-288, -48, 3)

julia> stride(vpv, 4) # I think this one is wrong; SubArray is implicitly assuming column-major parents
6
```

That last SubArray behavior can and should be tackled separately, but this gives me a bit of intuition here.

Reply:
Not sure I follow...

Reply:
We're adding a behavior here — I just want to make sure that it makes sense with respect to our existing behaviors and that there's a decent rationale for it.

Reply:
Yes I get that, I don't follow what you are proposing. (It seems to me that it doesn't really matter what the stride is outside the range)

Reply:
This was by-and-large just me talking out loud.  It's not a major thing since — as you say — it shouldn't matter.  Even if we use a value that's more likely to push someone out of bounds, we don't even check pointer accesses with `--check-bounds=yes`.  It _might_ be more likely that it'd help someone identify the problem, but that's quite the shot in the dark.

Upon reflection here I still do have a slight preference for `return sum(st .* size(A))` here simply because using `length` alone is a code smell in strides computations — it just looks wrong.  Indeed, it makes it clearer that we need to call `strides` in this branch (like Tim noted above).  But as a style thing, I'll leave the final decision to you.

Reply:
I'm happy with your suggestion .

Reply:
```suggestion
    return sum(st .* size(A))
```